apis/v1beta1/vspherevm_webhook.go

```diff
@@ -103,12 +103,16 @@ func (r *VSphereVM) ValidateUpdate(old runtime.Object) (admission.Warnings, erro
 	newVSphereVMSpec := newVSphereVM["spec"].(map[string]interface{})
 	oldVSphereVMSpec := oldVSphereVM["spec"].(map[string]interface{})
 
-	// allow changes to biosUUID, bootstrapRef, thumbprint
-	keys := []string{"biosUUID", "bootstrapRef", "thumbprint", "powerOffMode", "guestSoftPowerOffTimeout"}
-	// allow changes to os only if the old spec has empty OS field
+	// Allow changes to bootstrapRef, thumbprint, powerOffMode, guestSoftPowerOffTimeout.
+	keys := []string{"bootstrapRef", "thumbprint", "powerOffMode", "guestSoftPowerOffTimeout"}
+	// Allow changes to os only if the old spec has empty OS field.
 	if _, ok := oldVSphereVMSpec["os"]; !ok {
 		keys = append(keys, "os")
 	}
+	// Allow changes to biosUUID only if it is not already set.
+	if _, ok := oldVSphereVMSpec["biosUUID"]; !ok {
```

Comment:
I assume when the field is not set this entry does not exist and it's not an empty string, right?

Reply:
I know the test sort of confirms it, just want to double check

Reply:
Q: What about checking the typed object instead for an empty string? old.Spec.BiosUUID?
(just feels slightly better)